### DISCUSSION TITLE
Speed Improvements in Python 3.11

### DISCUSSION THREAD

**Discussion by kinghuang** [explosion/spaCy#11392] [Help: Other Questions] [answered]
I'm curious if anyone's tried spaCy on Python 3.11? The [faster-cpython project](https://github.com/faster-cpython/) is showing 10 to 60% faster performance on the `pyperformance` benchmarks. Given spaCy's heavy usage of CPython, I wonder if there'll be a noticeable speed improvement with Python 3.11 this fall?

**Comment by shadeMe** [comment, marked as answer]
We ran a couple of benchmarks a few months ago with one of the pre-release builds. At that time, we saw a small but noticeable improvement (between 5-10%) in some inference workloads. However, it must be noted that not all workloads saw an appreciable performance uplift, and external factors such as platform, OS, etc. play into this.

In any event, we hope to run more comprehensive benchmarks after the public release of Python 3.11.

**Reply by danieldk** [reply]
Just wanted to add that spaCy spends a relatively large amount of time in C (e.g. matrix multiplication) and Cython code. So the improvements are not as profound as you'd see in pure Python projects.

**Reply by kinghuang** [reply]
Cool, looking forward to even a small bump! :)
